File: 백준/Silver/28278. 스택 2/스택 2.py
import sys
input = sys.stdin.readline

N = int(input())

xList = list()

for _ in range(N):
    order = input()
    if order[0] == '1':
        num, x = map(int, order.split())
        xList.append(x)
    elif order[0] == '2':
        if len(xList) == 0:
            print(-1)
        else:
            print(xList[-1])
            xList.pop()
    elif order[0] == '3':
        print(len(xList))
    elif order[0] == '4':
        if len(xList) == 0:
            print(1)
        else:
            print(0) 
    elif order[0] == '5':
        if len(xList) == 0:
            print(-1)
        else:
            print(xList[-1])

File: 백준/Silver/28278. 스택 2/스택 2.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());
        Stack<Integer> xList = new Stack<>();

        for(int i = 0; i < N; i++){
            String order = br.readLine();
            if(order.charAt(0) == '1'){
                StringTokenizer st = new StringTokenizer(order);
                String A = st.nextToken();
                int X = Integer.parseInt(st.nextToken());
                xList.push(X);
            }else if(order.charAt(0) == '2'){
                if (xList.isEmpty()) {
                    System.out.println("-1");
                } else {
                    System.out.println(xList.pop());
                }
            }else if(order.charAt(0) == '3'){
                System.out.println(xList.size());
            }else if(order.charAt(0) == '4'){
                if (xList.isEmpty()) {
                    System.out.println(1);
                }else{
                    System.out.println(0);
                }
            }else if(order.charAt(0) == '5'){
                if (xList.isEmpty()) {
                    System.out.println(-1);
                }else{
                    System.out.println(xList.peek());
                }
            }
        }
        br.close();
    }
}